vignettes/irecist.Rmd

```diff
@@ -0,0 +1,858 @@
+    paste0(
+      "[", text, "]",
+      "(", url, ")"
+    )
+  )
+}
+dyn_link <- function(text,
+                     base_url,
+                     relative_url = "",
+                     # Change to TRUE when admiral adopts multiversion docs
+                     is_multiversion = FALSE,
+                     multiversion_default_ref = "main") {
+  url <- paste(base_url, relative_url, sep = "/")
+  if (is_multiversion) {
+    url <- paste(
+      base_url,
+      Sys.getenv("BRANCH_NAME", multiversion_default_ref),
+      relative_url,
+      sep = "/"
+    )
+  }
+  return(link(text, url))
+}
+# Other variables
+admiral_homepage <- "https://pharmaverse.github.io/admiral/cran-release"
+
+library(admiraldev)
+```
+
+# Introduction
+
+This article describes creating an `ADRS` ADaM with oncology endpoint parameters
+based on iRECIST. It shows a similar way of deriving the endpoints presented in 
+[Creating ADRS (Including Non-standard Endpoints)](adrs.html). Most of the endpoints 
+are derived by calling `admiral::derive_extreme_event()`. Thus the examples in this 
+vignette can also be used as a starting point for implementing response criteria other 
+than iRECIST 1.1,  e.g., RECIST 1.1 or International Myeloma Working Group (IMWG) 
+criteria for the diagnosis of multiple myeloma.
+
+For more information on the iRECIST guidelines user may visit
+https://recist.eortc.org/recist-1-1-2/
+
+Examples are currently presented and tested using `ADSL` (ADaM) and
+`RS` (SDTM) inputs. However, other domains could be used. The functions 
+and workflow could similarly be used to create an intermediary `ADEVENT` ADaM.
+
+**Note**: *All examples assume CDISC SDTM and/or ADaM format as input
+unless otherwise specified.*
+
+# Programming Workflow
+
+-   [Read in Data](#readdata)
+-   [Pre-processing of Input Records](#input)
+-   [Derive Confirmed Progressive Disease Parameter](#pd)
+-   [Derive Response Parameter](#rsp)
+-   [Derive Clinical Benefit Parameter](#cb)
+-   [Derive Best Overall Response Parameter](#bor)
+-   [Derive Response Parameters requiring Confirmation](#confirm)
+
+## Read in Data {#readdata}
+
+To start, all data frames needed for the creation of `ADRS` should be
+read into the environment. This will be a company specific process. Some
+of the data frames needed may be `ADSL`, `RS` and `TU`.
+
+For example purpose, the SDTM and ADaM datasets (based on CDISC Pilot
+test data)---which are included in `{pharmaversesdtm}` and `{pharmaverseadam}`---are used.
+
+```{r message=FALSE}
+library(admiral)
+library(admiralonco)
+library(dplyr)
+library(pharmaverseadam)
+library(pharmaversesdtm)
+library(lubridate)
+library(stringr)
+data("adsl")
+data("rs_onco_irecist")
+
+adsl_onco <- adsl
+rs <- rs_onco_irecist
+
+rs <- convert_blanks_to_na(rs)
+```
+
+```{r echo=FALSE}
+# select subjects from adsl such that there is one subject without RS data
+rs_subjects <- unique(rs$USUBJID)
+adsl_subjects <- unique(adsl_onco$USUBJID)
+adsl <- filter(
```

Comment:
echo = FALSE prevents code, definition of adsl in the vignette is unknown

Reply:
updated to remove adsl <- adsl_onco and similar code.